diff -Naur pwlib-1.8.4.orig/make/common.mak pwlib-1.8.4/make/common.mak
--- pwlib-1.8.4.orig/make/common.mak	2004-06-10 03:36:44.000000000 +0200
+++ pwlib-1.8.4/make/common.mak	2005-05-20 09:35:06.000000000 +0200
@@ -384,6 +384,7 @@
 	$(LD)   --split-by-reloc 65535 -r   $(OBJS) -o $@ 
 endif
 
+ifdef PWLIB_BUILD
 ifdef DEBUG
 
 ifneq (,$(wildcard $(PWLIBDIR)/src/ptlib/unix))
@@ -399,6 +400,7 @@
 endif
 
 endif
+endif
 
 CLEAN_FILES += $(TARGET)
 
diff -Naur pwlib-1.8.4.orig/make/gui.mak pwlib-1.8.4/make/gui.mak
--- pwlib-1.8.4.orig/make/gui.mak	2001-06-15 02:58:00.000000000 +0200
+++ pwlib-1.8.4/make/gui.mak	2005-05-20 09:35:06.000000000 +0200
@@ -64,11 +64,11 @@
 #
 
 
-include $(PWLIBDIR)/make/defaultgui.mak
+include $(PWLIBMAKEDIR)/defaultgui.mak
 
 ifneq (,$(GUI_TYPE))
 
-include $(PWLIBDIR)/make/$(GUI_TYPE).mak
+include $(PWLIBMAKEDIR)/$(GUI_TYPE).mak
 
 ifndef	GUI_SRC_NAME
 GUI_SRC_NAME	= $(GUI_TYPE)
diff -Naur pwlib-1.8.4.orig/make/ptbuildopts.mak.in pwlib-1.8.4/make/ptbuildopts.mak.in
--- pwlib-1.8.4.orig/make/ptbuildopts.mak.in	2004-11-17 11:13:14.000000000 +0100
+++ pwlib-1.8.4/make/ptbuildopts.mak.in	2005-05-20 09:35:06.000000000 +0200
@@ -2,7 +2,7 @@
 #
 
 ifndef PWLIBDIR
-PWLIBDIR   = @PWLIBDIR@
+PWLIBDIR   = @INSTALLPREFIX@
 endif
 
 MACHTYPE   = @MACHTYPE@
@@ -12,7 +12,6 @@
 STDCCFLAGS    += @STDCCFLAGS@
 STDCXXFLAGS   += @STDCXXFLAGS@
 OPTSTDCCFLAGS += @OPTSTDCCFLAGS@
-LDFLAGS	      += @LDFLAGS@
 ENDLDLIBS     += @ENDLDLIBS@
 
 HAS_OPENSSL	= @HAS_OPENSSL@
diff -Naur pwlib-1.8.4.orig/make/ptlib-config.in pwlib-1.8.4/make/ptlib-config.in
--- pwlib-1.8.4.orig/make/ptlib-config.in	2004-08-22 06:22:18.000000000 +0200
+++ pwlib-1.8.4/make/ptlib-config.in	2005-05-20 09:35:06.000000000 +0200
@@ -6,10 +6,9 @@
 
 VERSION="@PWLIB_VERSION@"
 CCFLAGS="@STDCCFLAGS@"
-LDFLAGS="@LDFLAGS@"
 LIBS="@ENDLDLIBS@"
 LIBDIR="@LIBDIR@"
-PWINSTDIR="@INSTALLPREFIX@/share/pwlib"
+PWINSTDIR="@INSTALLPREFIX@"
 MACHTYPE="@MACHTYPE@"
 OSTYPE="@OSTYPE@"
 OSRELEASE="@OSRELEASE@"
diff -Naur pwlib-1.8.4.orig/make/ptlib.mak pwlib-1.8.4/make/ptlib.mak
--- pwlib-1.8.4.orig/make/ptlib.mak	1998-12-02 03:37:31.000000000 +0100
+++ pwlib-1.8.4/make/ptlib.mak	2005-05-20 09:35:06.000000000 +0200
@@ -35,10 +35,13 @@
 #
 
 ifndef PWLIBDIR
-PWLIBDIR=$(HOME)/pwlib
+PWLIBDIR=/usr
+endif
+ifndef PWLIBMAKEDIR
+PWLIBMAKEDIR=/usr/share/pwlib/make
 endif
 
-include $(PWLIBDIR)/make/unix.mak
-include $(PWLIBDIR)/make/common.mak
+include $(PWLIBMAKEDIR)/unix.mak
+include $(PWLIBMAKEDIR)/common.mak
 
 # End of ptlib.mak
diff -Naur pwlib-1.8.4.orig/make/pwlib.mak pwlib-1.8.4/make/pwlib.mak
--- pwlib-1.8.4.orig/make/pwlib.mak	1998-12-02 03:39:53.000000000 +0100
+++ pwlib-1.8.4/make/pwlib.mak	2005-05-20 09:35:06.000000000 +0200
@@ -44,11 +44,14 @@
 #
 
 ifndef PWLIBDIR
-PWLIBDIR=$(HOME)/pwlib
+PWLIBDIR=/usr
+endif
+ifndef PWLIBMAKEDIR
+PWLIBMAKEDIR=/usr/share/pwlib/make
 endif
 
-include $(PWLIBDIR)/make/unix.mak
-include $(PWLIBDIR)/make/gui.mak
-include $(PWLIBDIR)/make/common.mak
+include $(PWLIBMAKEDIR)/unix.mak
+include $(PWLIBMAKEDIR)/gui.mak
+include $(PWLIBMAKEDIR)/common.mak
 
 # End of pwlib.mak
diff -Naur pwlib-1.8.4.orig/make/unix.mak pwlib-1.8.4/make/unix.mak
--- pwlib-1.8.4.orig/make/unix.mak	2005-05-20 09:36:08.000000000 +0200
+++ pwlib-1.8.4/make/unix.mak	2005-05-20 09:35:06.000000000 +0200
@@ -128,13 +128,16 @@
 #
 
 ifndef PWLIBDIR
-PWLIBDIR = $(HOME)/pwlib
+PWLIBDIR=/usr
+endif
+ifndef PWLIBMAKEDIR
+PWLIBMAKEDIR=/usr/share/pwlib/make
 endif
 
 ####################################################
 
 # include generated build options file, then include it
-include $(PWLIBDIR)/make/ptbuildopts.mak
+include $(PWLIBMAKEDIR)/ptbuildopts.mak
 
 ###############################################################################
 #
@@ -357,10 +360,6 @@
 # any -W overrides won't have any effect
 STDCCFLAGS += -Wall 
 
-ifdef RPM_OPT_FLAGS
-STDCCFLAGS	+= $(RPM_OPT_FLAGS)
-endif
-
 ifneq ($(OSTYPE),rtems)
 ifndef SYSINCDIR
 SYSINCDIR := /usr/include
@@ -405,7 +404,7 @@
 
 
 STATIC_LIBS	:= libstdc++.a libg++.a libm.a libc.a
-SYSLIBDIR	:= $(shell $(PWLIBDIR)/make/ptlib-config --libdir)
+SYSLIBDIR	:= $(shell $(PWLIBDIR)/bin/ptlib-config --libdir)
 
 endif # linux
 
@@ -851,7 +850,7 @@
 endif
 
 ifndef INSTALLLIB_DIR
-INSTALLLIB_DIR	= $(INSTALL_DIR)/lib
+INSTALLLIB_DIR	= $(subst $(PWLIBDIR),$(INSTALL_DIR),$(SYSLIBDIR))
 endif
 
 
@@ -878,7 +877,7 @@
 UNIX_SRC_DIR	= $(PWLIBDIR)/src/ptlib/unix
 endif
 
-PW_LIBDIR	= $(PWLIBDIR)/lib
+PW_LIBDIR	= $(SYSLIBDIR)
 
 # set name of the PT library
 PTLIB_BASE	= pt
@@ -926,7 +925,7 @@
 # define ESDDIR variables if installed
 ifdef  ESDDIR
 STDCCFLAGS	+= -I$(ESDDIR)/include -DUSE_ESD=1
-ENDLDLIBS	+= $(ESDDIR)/lib/libesd.a  # to avoid name conflicts
+ENDLDLIBS	+= -lesd
 HAS_ESD		= 1
 endif
 
diff -Naur pwlib-1.8.4.orig/Makefile.in pwlib-1.8.4/Makefile.in
--- pwlib-1.8.4.orig/Makefile.in	2004-06-24 14:46:48.000000000 +0200
+++ pwlib-1.8.4/Makefile.in	2005-05-20 09:35:06.000000000 +0200
@@ -247,26 +247,16 @@
 		     $(DESTDIR)$(PREFIX)/include/ptlib \
                      $(DESTDIR)$(PREFIX)/include/ptlib/unix/ptlib \
                      $(DESTDIR)$(PREFIX)/include/ptclib \
-                     $(DESTDIR)$(PREFIX)/share/pwlib/make ; \
+                     $(DESTDIR)$(PREFIX)/share/pwlib/make \
+		     $(DESTDIR)$(PREFIX)/share/man/man1 ;\
 		do mkdir -p $$dir ; chmod 755 $$dir ; \
 	done )
 	$(INSTALL) -m 444 lib/$(PTLIB_FILE).@MAJOR_VERSION@.@MINOR_VERSION@.@BUILD_NUMBER@ $(DESTDIR)$(LIBDIR)
 	(cd $(DESTDIR)$(LIBDIR) ; \
-		rm -f  $(PTLIB_FILE).@MAJOR_VERSION@.@MINOR_VERSION@ ; \
-		ln -sf $(PTLIB_FILE).@MAJOR_VERSION@.@MINOR_VERSION@.@BUILD_NUMBER@ $(PTLIB_FILE).@MAJOR_VERSION@.@MINOR_VERSION@ \
-	)
-	(cd $(DESTDIR)$(LIBDIR) ; \
-		rm -f  $(PTLIB_FILE).@MAJOR_VERSION@ ; \
-		ln -sf $(PTLIB_FILE).@MAJOR_VERSION@.@MINOR_VERSION@.@BUILD_NUMBER@ $(PTLIB_FILE).@MAJOR_VERSION@ \
-	)
-	(cd $(DESTDIR)$(LIBDIR) ; \
 		rm -f  $(PTLIB_FILE) ; \
 		ln -sf $(PTLIB_FILE).@MAJOR_VERSION@.@MINOR_VERSION@.@BUILD_NUMBER@ $(PTLIB_FILE) \
 	)
-	(cd $(DESTDIR)$(LIBDIR); \
-		rm -f  libpt.@SHAREDLIBEXT@ ; \
-		ln -sf $(PTLIB_FILE) libpt.@SHAREDLIBEXT@ \
-	)
+	install -m644 lib/libpt.a $(DESTDIR)$(LIBDIR)
 ifeq (1, $(HAS_PLUGINS))
 	cd plugins/pwlib/device/ ; \
 	( for dir in ./* ;\
@@ -277,8 +267,8 @@
 		done ); \
 	done )
 endif
-	$(INSTALL) -m 444 include/ptlib.h                $(DESTDIR)$(PREFIX)/include
-	$(INSTALL) -m 444 include/ptbuildopts.h          $(DESTDIR)$(PREFIX)/include
+	$(INSTALL) -m 444 include/ptlib.h $(DESTDIR)$(PREFIX)/include
+	$(INSTALL) -m 444 include/ptbuildopts.h  $(DESTDIR)$(PREFIX)/include
 	(for fn in include/ptlib/*.h include/ptlib/*.inl; \
 		do $(INSTALL) -m 444 $$fn $(DESTDIR)$(PREFIX)/include/ptlib; \
 	done)
