--- pwlib-1.10.10/src/ptlib/common/osutils.cxx.orig	2007-11-28 16:50:46.000000000 +0000
+++ pwlib-1.10.10/src/ptlib/common/osutils.cxx	2007-11-28 17:01:24.000000000 +0000
@@ -2145,9 +2145,9 @@
 void PThread::SetThreadName(const PString & name)
 {
   if (name.IsEmpty())
-    threadName = psprintf("%s:%08x", GetClass(), (INT)this);
+    threadName = psprintf("%s:%08x", GetClass(), this);
   else
-    threadName = psprintf(name, (INT)this);
+    threadName = psprintf(name, this);
 
 #if defined(_DEBUG) && defined(_MSC_VER)
   if (threadId) {       // make thread name known to debugger
