diff -aurN pwlib.orig/Makefile.in pwlib/Makefile.in
--- pwlib.orig/Makefile.in	2003-05-22 14:13:44.000000000 +0200
+++ pwlib/Makefile.in	2003-07-12 21:04:13.000000000 +0200
@@ -164,22 +164,23 @@
 
 install:
 	chmod a+r lib/*@SHAREDLIBEXT@*
-	cp -df lib/*@SHAREDLIBEXT@* $(PREFIX)/lib/
-	cp -rf include/ptclib/ $(PREFIX)/include/
-	cp -rf include/ptlib/ $(PREFIX)/include/
-	cp -rf include/ptbuildopts.h $(PREFIX)/include/
-	cp -rf include/ptlib.h $(PREFIX)/include/
-	cp tools/asnparser/obj*/asnparser $(PREFIX)/bin/
-	cp make/ptlib-config $(PREFIX)/bin/
-	mkdir -p $(PREFIX)/share/pwlib/
-	cp -rf make $(PREFIX)/share/pwlib/
-	ln -s $(PREFIX)/lib/$(PTLIB_FILE) $(PREFIX)/lib/libpt.@SHAREDLIBEXT@
-	chmod -R a+r $(PREFIX)/include/ptlib $(PREFIX)/include/ptclib $(PREFIX)/include/ptlib.h $(PREFIX)/share/pwlib $(PREFIX)/include/ptbuildopts.h $(PREFIX)/include/ptlib.h
-	chmod a+rx $(PREFIX)/include/ptlib $(PREFIX)/include/ptlib/unix $(PREFIX)/include/ptlib/unix/ptlib $(PREFIX)/include/ptclib $(PREFIX)/bin/ptlib-config $(PREFIX)/share/pwlib/make/
-
+	install -d -m755 $(DESTDIR)$(PREFIX)/{lib,include/{ptclib,ptlib/unix/ptlib},bin,share/pwlib/make,share/man/man1}
+	cp -df lib/*@SHAREDLIBEXT@* $(DESTDIR)$(PREFIX)/lib/
+	install -m644 lib/libpt.a $(DESTDIR)$(PREFIX)/lib
+	install -m644 include/ptclib/*.h $(DESTDIR)$(PREFIX)/include/ptclib
+	install -m644 include/ptlib/*.{h,inl} $(DESTDIR)$(PREFIX)/include/ptlib
+	install -m644 include/ptlib/unix/ptlib/*.{h,inl} $(DESTDIR)$(PREFIX)/include/ptlib/unix/ptlib
+	install -m644 include/ptbuildopts.h $(DESTDIR)$(PREFIX)/include
+	install -m644 include/ptlib.h $(DESTDIR)$(PREFIX)/include
+	install -m755 tools/asnparser/obj*/asnparser $(DESTDIR)$(PREFIX)/bin
+	install -m755 make/ptlib-config $(DESTDIR)$(PREFIX)/bin
+	install -m644 make/*.mak $(DESTDIR)$(PREFIX)/share/pwlib/make
+	chmod a+rx lib/*@SHAREDLIBEXT@*
+	install -m644 tools/asnparser/asnparser.1 $(DESTDIR)$(PREFIX)/share/man/man1
+	
 uninstall:
-	rm -rf $(PREFIX)/include/ptlib $(PREFIX)/include/ptclib $(PREFIX)/include/ptlib.h $(PREFIX)/include/ptbuildopts.h $(PREFIX)/share/pwlib/ 
-	rm -f $(PREFIX)/lib/$(PTLIB_FILE)* $(PREFIX)/lib/libpt.so $(PREFIX)/bin/asnparser $(PREFIX)/bin/ptlib-config
+	rm -rf $(DESTDIR)$(PREFIX)/include/ptlib $(DESTDIR)$(PREFIX)/include/ptclib $(DESTDIR)$(PREFIX)/include/ptlib.h $(DESTDIR)$(PREFIX)/include/ptbuildopts.h $(DESTDIR)$(PREFIX)/share/pwlib/ 
+	rm -f $(DESTDIR)$(PREFIX)/lib/$(PTLIB_FILE)* $(DESTDIR)$(PREFIX)/lib/libpt.so $(DESTDIR)$(PREFIX)/bin/asnparser $(DESTDIR)$(PREFIX)/bin/ptlib-config
 
 
 # End of Makefile.in
