diff -aurN pwlib.orig/Makefile.in pwlib/Makefile.in
--- pwlib.orig/Makefile.in	2003-05-22 14:13:44.000000000 +0200
+++ pwlib/Makefile.in	2003-07-12 21:04:13.000000000 +0200
@@ -164,22 +164,35 @@
 
 install:
 	chmod a+r lib/*@SHAREDLIBEXT@*
-	cp -df lib/*@SHAREDLIBEXT@* $(PREFIX)/lib/
-	cp -rf include/ptclib/ $(PREFIX)/include/
-	cp -rf include/ptlib/ $(PREFIX)/include/
-	cp -rf include/ptbuildopts.h $(PREFIX)/include/
-	cp -rf include/ptlib.h $(PREFIX)/include/
-	cp tools/asnparser/obj*/asnparser $(PREFIX)/bin/
-	cp make/ptlib-config $(PREFIX)/bin/
-	mkdir -p $(PREFIX)/share/pwlib/
-	cp -rf make $(PREFIX)/share/pwlib/
-	ln -s $(PREFIX)/lib/$(PTLIB_FILE) $(PREFIX)/lib/libpt.@SHAREDLIBEXT@
-	chmod -R a+r $(PREFIX)/include/ptlib $(PREFIX)/include/ptclib $(PREFIX)/include/ptlib.h $(PREFIX)/share/pwlib $(PREFIX)/include/ptbuildopts.h $(PREFIX)/include/ptlib.h
-	chmod a+rx $(PREFIX)/include/ptlib $(PREFIX)/include/ptlib/unix $(PREFIX)/include/ptlib/unix/ptlib $(PREFIX)/include/ptclib $(PREFIX)/bin/ptlib-config $(PREFIX)/share/pwlib/make/
-
+	cp -df lib/*@SHAREDLIBEXT@* $(DESTDIR)$(PREFIX)/lib/
+	cp lib/libpt.a $(DESTDIR)$(PREFIX)/lib/
+	cp -rf include/ptclib/ $(DESTDIR)$(PREFIX)/include/
+	cp -rf include/ptlib/ $(DESTDIR)$(PREFIX)/include/
+	cp -rf include/ptbuildopts.h $(DESTDIR)$(PREFIX)/include/
+	cp -rf include/ptlib.h $(DESTDIR)$(PREFIX)/include/
+	cp tools/asnparser/obj*/asnparser $(DESTDIR)$(PREFIX)/bin/
+	cp make/ptlib-config $(DESTDIR)$(PREFIX)/bin/
+	mkdir -p $(DESTDIR)$(PREFIX)/share/pwlib/
+	cp -rf make $(DESTDIR)$(PREFIX)/share/pwlib/
+	chmod -R a+r $(DESTDIR)$(PREFIX)/include/ptlib $(DESTDIR)$(PREFIX)/include/ptclib $(DESTDIR)$(PREFIX)/include/ptlib.h $(DESTDIR)$(PREFIX)/share/pwlib $(DESTDIR)$(PREFIX)/include/ptbuildopts.h $(DESTDIR)$(PREFIX)/include/ptlib.h
+	chmod a+rx $(DESTDIR)$(PREFIX)/include/ptlib $(DESTDIR)$(PREFIX)/include/ptlib/unix $(DESTDIR)$(PREFIX)/include/ptlib/unix/ptlib $(DESTDIR)$(PREFIX)/include/ptclib $(DESTDIR)$(PREFIX)/bin/ptlib-config $(DESTDIR)$(PREFIX)/share/pwlib/make/
+	chmod a+r lib/*@SHAREDLIBEXT@*
+	cp -df lib/*@SHAREDLIBEXT@* $(DESTDIR)$(PREFIX)/lib/
+	cp -rf include/ptclib/ $(DESTDIR)$(PREFIX)/include/
+	cp -rf include/ptlib/ $(DESTDIR)$(PREFIX)/include/
+	cp -rf include/ptbuildopts.h $(DESTDIR)$(PREFIX)/include/
+	cp -rf include/ptlib.h $(DESTDIR)$(PREFIX)/include/
+	cp tools/asnparser/obj*/asnparser $(DESTDIR)$(PREFIX)/bin/
+	cp make/ptlib-config $(DESTDIR)$(PREFIX)/bin/
+	mkdir -p $(DESTDIR)$(PREFIX)/share/pwlib/
+	cp -rf make $(DESTDIR)$(PREFIX)/share/pwlib/
+	chmod -R a+r $(DESTDIR)$(PREFIX)/include/ptlib $(DESTDIR)$(PREFIX)/include/ptclib $(DESTDIR)$(PREFIX)/include/ptlib.h $(DESTDIR)$(PREFIX)/share/pwlib $(DESTDIR)$(PREFIX)/include/ptbuildopts.h $(DESTDIR)$(PREFIX)/include/ptlib.h
+	chmod a+rx $(DESTDIR)$(PREFIX)/include/ptlib $(DESTDIR)$(PREFIX)/include/ptlib/unix $(DESTDIR)$(PREFIX)/include/ptlib/unix/ptlib $(DESTDIR)$(PREFIX)/include/ptclib $(DESTDIR)$(PREFIX)/bin/ptlib-config $(DESTDIR)$(PREFIX)/share/pwlib/make/
+	cp tools/asnparser/asnparser.1 $(DESTDIR)$(PREFIX)/share/man/man1
+	
 uninstall:
-	rm -rf $(PREFIX)/include/ptlib $(PREFIX)/include/ptclib $(PREFIX)/include/ptlib.h $(PREFIX)/include/ptbuildopts.h $(PREFIX)/share/pwlib/ 
-	rm -f $(PREFIX)/lib/$(PTLIB_FILE)* $(PREFIX)/lib/libpt.so $(PREFIX)/bin/asnparser $(PREFIX)/bin/ptlib-config
+	rm -rf $(DESTDIR)$(PREFIX)/include/ptlib $(DESTDIR)$(PREFIX)/include/ptclib $(DESTDIR)$(PREFIX)/include/ptlib.h $(DESTDIR)$(PREFIX)/include/ptbuildopts.h $(DESTDIR)$(PREFIX)/share/pwlib/ 
+	rm -f $(DESTDIR)$(PREFIX)/lib/$(PTLIB_FILE)* $(DESTDIR)$(PREFIX)/lib/libpt.so $(DESTDIR)$(PREFIX)/bin/asnparser $(DESTDIR)$(PREFIX)/bin/ptlib-config
 
 
 # End of Makefile.in
