diff -urN pwlib.org/make/common.mak pwlib/make/common.mak
--- pwlib.org/make/common.mak	Tue Jan  8 00:03:22 2002
+++ pwlib/make/common.mak	Tue Jan  8 00:41:21 2002
@@ -295,6 +295,7 @@
 endif
 	$(CPLUS) -o $@ $(CFLAGS) $(LDFLAGS) $(OBJS) $(LDLIBS) $(ENDLDLIBS) $(ENDLDFLAGS)
 
+ifdef PWLIB_BUILD
 ifdef DEBUG
 
 $(PW_LIBDIR)/$(PTLIB_FILE):
@@ -312,6 +313,7 @@
 	$(MAKE) -C $(PWLIBDIR)/src/pwlib/$(PWLIB_GUI) opt
 
 endif
+endif
 
 CLEAN_FILES += $(TARGET)
 
diff -urN pwlib.org/make/ptlib.mak pwlib/make/ptlib.mak
--- pwlib.org/make/ptlib.mak	Tue Jan  8 00:03:22 2002
+++ pwlib/make/ptlib.mak	Tue Jan  8 00:41:21 2002
@@ -35,10 +35,15 @@
 #
 
 ifndef PWLIBDIR
-PWLIBDIR=$(HOME)/pwlib
+PWLIBDIR=@prefix@
 endif
 
+ifdef PWLIB_BUILD
 include $(PWLIBDIR)/make/unix.mak
 include $(PWLIBDIR)/make/common.mak
+else
+include @makdir@/unix.mak
+include @makdir@/common.mak
+endif
 
 # End of ptlib.mak
diff -urN pwlib.org/make/pwlib.mak pwlib/make/pwlib.mak
--- pwlib.org/make/pwlib.mak	Tue Jan  8 00:03:22 2002
+++ pwlib/make/pwlib.mak	Tue Jan  8 00:41:21 2002
@@ -44,11 +44,17 @@
 #
 
 ifndef PWLIBDIR
-PWLIBDIR=$(HOME)/pwlib
+PWLIBDIR=@prefix@
 endif
 
+ifdef PWLIB_BUILD
 include $(PWLIBDIR)/make/unix.mak
 include $(PWLIBDIR)/make/gui.mak
 include $(PWLIBDIR)/make/common.mak
+else
+include @makdir@/unix.mak
+include @makdir@/gui.mak
+include @makdir@/common.mak
+endif
 
 # End of pwlib.mak
diff -urN pwlib.org/make/unix.mak pwlib/make/unix.mak
--- pwlib.org/make/unix.mak	Tue Jan  8 00:03:22 2002
+++ pwlib/make/unix.mak	Tue Jan  8 00:41:21 2002
@@ -328,7 +328,7 @@
 
 
 ifndef PWLIBDIR
-PWLIBDIR = $(HOME)/pwlib
+PWLIBDIR = @prefix@
 endif
 
 ###############################################################################
